lib/routes/emails.js

```diff
@@ -470,6 +483,16 @@ module.exports = (log, db, mailer, config, customs, push) => {
             () => reply({}),
             reply
           )
+
+        function isServerVerified(code, uid, requestSignature) {
+          const signature = crypto.createHmac('sha256', SERVER_VERIFICATION_KEY)
+          .update([
+            code,
+            uid
+          ].join('\n'))
```

Comment:
In general, it's dangerous to have signed tokens that live forever.  If we happened to leak one of these tokens accidentally, then an attacker could use that to bypass rate-limiting for this endpoint on an indefinite basis, until we changed the signing key.  I think we should somehow ensure that these signatures expire after a couple of minutes.

There are a couple of ways we could put an expiry timestamp in here, but I wonder if we're going to be better off pushing this down into a library, and using e.g. a JWT rather than rolling our own signature here.  @vladikoff any thoughts on that, based on your recent experience with such libraries?